apps/meteor/server/models/raw/Messages.ts

```diff
@@ -1355,8 +1355,34 @@ export class MessagesRaw extends BaseRaw<IMessage> implements IMessagesModel {
 			query.tcount = { $exists: false };
 		}
 
-		if (!limit) {
-			const count = (await this.deleteMany(query)).deletedCount;
+		return (
+			await this.find(query, {
+				projection: {
+					_id: 1,
+				},
+				limit,
+			}).toArray()
+		).map(({ _id }) => _id);
+	}
+
+	async removeByIdPinnedTimestampLimitAndUsers(
+		rid: string,
+		pinned: boolean,
```

Comment:
I believe we should change this parameter to `ignore` or `exclude` pinned since it looks like the behavior intended here. Or the below filter `query.pinned = { $ne: true };` could be wrong, I'm not sure which is the expected behavior though.

Reply:
great suggestion, agreed. I changed it to `ignorePinned` (in the `find` also)